Picklist: Transfering the items from source to target causing the script error and browser crash for more than 4000 items on IE browser

# IMPORTANT 
- !!! If you open an issue, fill every item. Otherwise the issue might be closed as invalid. !!!

- Please use the naming convention for the title:
  ${component}: ${title}
  Example:
  SelectOneMenu: Converter not called
- Before you open an issue, test it with the current/newest version.
- Try to find an explanation to your problem by yourself, by simply debugging. This will help us to solve your issue 10x faster
- Clone this repository https://github.com/primefaces/primefaces-test.git in order to reproduce your problem, you'll have better chance to receive an answer and a solution.
- Otherwise the example must be as small and simple as possible! It must be runnable without any other dependencies (like Spring,..., or project/company internal classes)!
- Feel free to provide a PR (Primefaces is an open-source project, any fixes or improvements are welcome.)

## 1) Environment
- PrimeFaces version: 6.2.9
- Does it work on the newest released PrimeFaces version? Version?
- Does it work on the newest sources in GitHub? (Build by source -> https://github.com/primefaces/primefaces/wiki/Building-From-Source)
- Application server + version:
- Affected browsers: IE/Morzilla Firefox 52.0(Sometimes)

## 2) Expected behavior : While transferring the items from source to target should not lag and throw script error. 
...

## 3) Actual behavior : While transferring the items from source to target on IE browser, screen is lagging and throwing the script error.We have used a UI blocker ontransfer event(onstart(),onComplete()), that event is also getting called after some seconds.That means UI is getting blocked after some seconds. But In Firefox UI is getting blocked immediately.


..

## 4) Steps to reproduce
Add more than 4000 items(In Source and Target) in the picklist (primeface version 6.2.9).
Search for User. 
click on User.
Drag and Drop(or Click on the Arrow) him to the other side.
The Screen will lag and if you click on the screen anywhere the script error will appear on screen(See Screenshot)

.. 

## 5) Sample XHTML
<p:pickList id="pickList" value="#{globalAuthorsActivation.users}" var="users" widgetVar="pick" effect="none"
				itemLabel="#{users}" itemValue="#{users}" showSourceFilter="true" showTargetFilter="true" filterMatchMode="contains" showCheckbox="true">
					<f:facet name="sourceCaption">#{msg['facet.ActiveUsers']}</f:facet>
        			<f:facet name="targetCaption">#{msg['facet.GlobalUsers']}</f:facet>
        			<p:ajax event="transfer" listener="#{globalAuthorsActivation.handleTransfer}" update="pickList,toUpdate" onstart="PF('bui').show()" oncomplete="PF('bui').hide()"/>
					<p:ajax event="select" listener="#{globalAuthorsActivation.onSelect}" update="toUpdate" onstart="PF('bui').show()" oncomplete="PF('bui').hide()"/>
					<p:ajax event="select" listener="#{globalAuthorsActivation.onSelect}" update="toUpdate" onstart="PF('bui').show()" oncomplete="PF('bui').hide()"/>
					<p:ajax event="unselect" listener="#{globalAuthorsActivation.onUnselect}" update="toUpdate" onstart="PF('bui').show()" oncomplete="PF('bui').hide()"/>
							
					 <p:column style="width:10%">
					 		<div onclick="onClickPickList('#{users}')">
								<h:outputLabel id="img_#{users}" styleClass="ui-user"/>
							</div>								
					</p:column>
					<p:column style="width:90%">
							<div onclick="onClickPickList('#{users}')">
								<h:outputLabel id="txt_#{users}" value="#{users}"/>
							</div>
					</p:column>
</p:pickList>

..

## 6) Sample bean
package com.wissen.admin.model;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashSet;
import java.util.Iterator;
import java.util.List;
import java.util.Map;
import java.util.Map.Entry;
import java.util.ResourceBundle;
import java.util.Set;
import java.util.TreeMap;

import javax.faces.application.FacesMessage;
import javax.faces.bean.ManagedBean;
import javax.faces.bean.ViewScoped;
import javax.faces.context.FacesContext;
import javax.faces.event.AjaxBehaviorEvent;

import org.apache.log4j.Logger;
import org.primefaces.event.SelectEvent;
import org.primefaces.event.TransferEvent;
import org.primefaces.event.UnselectEvent;
import org.primefaces.model.DualListModel;

import com.documentum.fc.client.IDfSession;
import com.wissen.admin.api.AdminAPI;
import com.wissen.admin.exception.AdminException;
import com.wissen.admin.util.AdminConstants;
import com.wissen.factory.WissenswerkDAOFactory;
import com.wissen.login.api.WissenswerkSessionManager;
import com.wissen.util.FacesMessageUtil;
import com.wissen.util.WissenswerkConstants;

@ManagedBean
@ViewScoped
public class GlobalAuthorsActivation {

	/**
	 * Logger Object to print log
	 */
	private static Logger logger = Logger
			.getLogger(WissenswerkConstants.LOGGER_NAME);

	private List<AreaRoleBO> areaList = new ArrayList<AreaRoleBO>();

	private List<String> userName;

	private String userNameValue = "";
	private List<String> globalGroup = new ArrayList<String>();

	private static List<String> userList = new ArrayList<String>();
	private String emptymessage = "";
	private boolean showPanel = false;

	private DualListModel<String> users = new DualListModel<String>(
			Collections.<String> emptyList(), Collections.<String> emptyList());

	private String selectedUserType = "";
	private String userTypeFilterList;
	private Map<String, String> userTypeFilterLists;
	private Set<String> usersSource = new HashSet<String>();
	private Set<String> usersTarget = new HashSet<String>();
	List<DocTypeRoleBO> docTypeList = new ArrayList<DocTypeRoleBO>();
	private boolean panelShow = false;

	public List<AreaRoleBO> getAreaList() {
		return areaList;
	}

	public void setAreaList(List<AreaRoleBO> areaList) {
		this.areaList = areaList;
	}

	public List<String> getUserName() {
		return userName;
	}

	public void setUserName(List<String> userName) {
		this.userName = userName;
	}

	public String getUserNameValue() {
		return userNameValue;
	}

	public void setUserNameValue(String userNameValue) {
		this.userNameValue = userNameValue;
	}

	public List<String> getGlobalGroup() {
		return globalGroup;
	}

	public void setGlobalGroup(List<String> globalGroup) {
		this.globalGroup = globalGroup;
	}

	public static List<String> getUserList() {
		return userList;
	}

	public static void setUserList(List<String> userList) {
		GlobalAuthorsActivation.userList = userList;
	}

	public String getEmptymessage() {
		return emptymessage;
	}

	public void setEmptymessage(String emptymessage) {
		this.emptymessage = emptymessage;
	}

	public boolean isShowPanel() {
		return showPanel;
	}

	public void setShowPanel(boolean showPanel) {
		this.showPanel = showPanel;
	}

	public DualListModel<String> getUsers() {
		return users;
	}

	public void setUsers(DualListModel<String> users) {
		this.users = users;
	}

	public String getSelectedUserType() {
		return selectedUserType;
	}

	public void setSelectedUserType(String selectedUserType) {
		this.selectedUserType = selectedUserType;
	}

	public String getUserTypeFilterList() {
		return userTypeFilterList;
	}

	public void setUserTypeFilterList(String userTypeFilterList) {
		this.userTypeFilterList = userTypeFilterList;
	}

	public Map<String, String> getUserTypeFilterLists() {
		return userTypeFilterLists;
	}

	public void setUserTypeFilterLists(Map<String, String> userTypeFilterLists) {
		this.userTypeFilterLists = userTypeFilterLists;
	}

	public boolean isPanelShow() {
		return panelShow;
	}

	public void setPanelShow(boolean panelShow) {
		this.panelShow = panelShow;
	}

	public List<DocTypeRoleBO> getDocTypeList() {
		return docTypeList;
	}

	public void setDocTypeList(List<DocTypeRoleBO> docTypeList) {
		this.docTypeList = docTypeList;
	}

	public void init() {
		final String lBuf = "com.wissen.admin.model.GlobalAuthorsActivation.init() :: ";
		logger.info(lBuf + WissenswerkConstants.INFO_ENTERED_METHOD);
		GlobalAuthorsActivation.userList.clear();
		showPanel = false;
		try {
			IDfSession session = WissenswerkSessionManager.instance()
					.getWEBSession();

			AdminAPI adminApi = new AdminAPI();
			String superUserID = WissenswerkDAOFactory
					.getmWissenswerkContextConfig()
					.getString(WissenswerkConstants.DOCBASE_SUPERUSER_NAME)
					.trim();
			String superUserName = adminApi.getFullNameByUserID(superUserID,
					session);

			usersSource = new HashSet<String>(adminApi.getUsersAllByGroupName(
					"grp_wissen_global_users", session));

			if (usersSource == null) {
				usersSource = new HashSet<String>();
			}
			if (usersTarget == null) {
				usersTarget = new HashSet<String>();
			}

			userTypeFilterLists = new TreeMap<String, String>();
			userTypeFilterLists.put(
					getResourceBundleMessage("user.global.reader"),
					AdminConstants.grp_wissen_global_read);
			userTypeFilterLists.put(
					getResourceBundleMessage("user.global.editor"),
					AdminConstants.grp_wissen_global_edit);
			userTypeFilterLists.put(
					getResourceBundleMessage("Admin.globalAreaCoordinator"),
					AdminConstants.grp_wissen_global_area_co);
			userTypeFilterLists.put(
					getResourceBundleMessage("Admin.globalReconCoordinator"),
					AdminConstants.grp_wissen_global_recon_co);
			userTypeFilterLists.put(
					getResourceBundleMessage("facet.GlobalAuthors"),
					AdminConstants.grp_wissen_global_authors);
			userTypeFilterLists.put(
					getResourceBundleMessage("global.administrator"),
					AdminConstants.grp_wissen_global_admin);
			System.out.println("GLOBAL USERS");

			if (usersSource.contains(session.getLoginUserName())) {
				usersSource.remove(session.getLoginUserName());
			}
			usersSource.remove(superUserName);
			
			List<String> sortedUsersSource = new ArrayList<String>(usersSource);
			Collections.sort(sortedUsersSource);
			
			List<String> sortedUsersTarget = new ArrayList<String>(usersTarget);
			Collections.sort(sortedUsersTarget);
			sortedUsersTarget.clear();
			users = new DualListModel<String>(
					new ArrayList<String>(sortedUsersSource), new ArrayList<String>(
							sortedUsersTarget));
			docTypeList = new ArrayList<DocTypeRoleBO>();
			globalGroup = new ArrayList<String>();
			this.selectedUserType = "";
		} catch (AdminException exp) {
			logger.error(lBuf + exp);
		} catch (Exception exp) {
			logger.error(lBuf + exp);
		} finally {
			logger.info(lBuf + WissenswerkConstants.INFO_EXITED_METHOD);
		}
	}

	public void activeUsers() {
		final String lBuf = "com.wissen.admin.model.GlobalAuthorsActivation.activeUsers() :: ";
		logger.info(lBuf + WissenswerkConstants.INFO_ENTERED_METHOD);
		GlobalAuthorsActivation.userList.clear();
		showPanel = false;
		try {

			if (!checkUserTypeSelected()) {

				IDfSession session = WissenswerkSessionManager.instance()
						.getWEBSession();

				AdminAPI adminApi = new AdminAPI();
				String superUserID = WissenswerkDAOFactory
						.getmWissenswerkContextConfig()
						.getString(WissenswerkConstants.DOCBASE_SUPERUSER_NAME)
						.trim();
				String superUserName = adminApi.getFullNameByUserID(
						superUserID, session);

				List<String> usersSourceFinal = new ArrayList<String>();

				usersSourceFinal.addAll(users.getSource());

				List<String> usersTargetFinal = new ArrayList<String>();

				usersTargetFinal.addAll(users.getTarget());
				if (AdminConstants.grp_wissen_global_admin
						.equals(this.selectedUserType)) {
					usersTargetFinal.add(session.getLoginUserName());
					usersTargetFinal.add(superUserName);
				}
				usersSourceFinal.add(session.getLoginUserName());
				usersSourceFinal.add(superUserName);
				adminApi.adminProcess(null, usersTargetFinal, null,
						this.selectedUserType, session);
				System.out.println("3ndsource :  " + users.getSource()
						+ " 3ndtarget :" + users.getTarget());
				String successfull = FacesMessageUtil.getString(
						"message.Successful", FacesMessageUtil.getLocale());
				addMessage(successfull, FacesMessageUtil.getString(
						"info.alloctionDone", FacesMessageUtil.getLocale()));
			}
		} catch (AdminException exp) {
			logger.error(lBuf + exp);
		} catch (Exception exp) {
			logger.error(lBuf + exp);
		} finally {
			logger.info(lBuf + WissenswerkConstants.INFO_EXITED_METHOD);
		}
	}

	private boolean checkUserTypeSelected() {
		if (this.selectedUserType == null || this.selectedUserType.isEmpty()) {
			addMessage(getResourceBundleMessage("emptyField.message"),
					FacesMessageUtil.getString("error.empty.username",
							FacesMessageUtil.getLocale()));
			return true;
		}

		return false;

	}

	public void addMessage(String summary, String detail) {
		FacesMessage message = new FacesMessage(FacesMessage.SEVERITY_INFO,
				summary, detail);
		FacesContext.getCurrentInstance().addMessage(null, message);
	}

	public void handleSiteTypeFilterChange(AjaxBehaviorEvent event) {
		final String lBuf = "com.wissen.admin.model.GlobalAuthorsActivation.handleSiteTypeFilterChange() :: ";
		logger.info(lBuf + WissenswerkConstants.INFO_ENTERED_METHOD);
		GlobalAuthorsActivation.userList.clear();
		showPanel = false;
		try {
			IDfSession session = WissenswerkSessionManager.instance()
					.getWEBSession();

			AdminAPI adminApi = new AdminAPI();
			String superUserID = WissenswerkDAOFactory
					.getmWissenswerkContextConfig()
					.getString(WissenswerkConstants.DOCBASE_SUPERUSER_NAME)
					.trim();
			String superUserName = adminApi.getFullNameByUserID(superUserID,
					session);

			usersTarget = new HashSet<String>(adminApi.getUsersAllByGroupName(
					this.selectedUserType, session));

			if (usersTarget == null) {
				usersTarget = new HashSet<String>();
			}

			if (AdminConstants.grp_wissen_global_admin
					.equals(this.selectedUserType)) {
				if (usersTarget.contains(session.getLoginUserName())) {
					usersTarget.remove(session.getLoginUserName());
				}
				usersTarget.remove(superUserName);
			}
			
			List<String> sortedUsersSource = new ArrayList<String>(usersSource);
			Collections.sort(sortedUsersSource);
			
			List<String> sortedUsersTarget = new ArrayList<String>(usersTarget);
			Collections.sort(sortedUsersTarget);
			
			users = new DualListModel<String>(
					new ArrayList<String>(sortedUsersSource), new ArrayList<String>(
							sortedUsersTarget));

		} catch (AdminException exp) {
			logger.error(lBuf + exp);
		} catch (Exception exp) {
			logger.error(lBuf + exp);
		} finally {
			logger.info(lBuf + WissenswerkConstants.INFO_EXITED_METHOD);
		}
	}

	@SuppressWarnings("unchecked")
	public void handleTransfer(TransferEvent event) {
		GlobalAuthorsActivation.userList.clear();
		showPanel = false;
		if (event.isAdd()) {
			{
				usersTarget.addAll((Collection<? extends String>) event
						.getItems());
			}

		}
		if (event.isRemove()) {

			usersTarget.removeAll((Collection<? extends String>) event
					.getItems());
		}
		
		List<String> sortedUsersSource = new ArrayList<String>(usersSource);
		Collections.sort(sortedUsersSource);
		
		List<String> sortedUsersTarget = new ArrayList<String>(usersTarget);
		Collections.sort(sortedUsersTarget);
		users = new DualListModel<String>(new ArrayList<String>(sortedUsersSource),
				new ArrayList<String>(sortedUsersTarget));

	}

	/**
	 * 
	 * @param key
	 * @return
	 */
	public String getResourceBundleMessage(String key) {
		FacesContext context = FacesContext.getCurrentInstance();
		ResourceBundle bundle = context.getApplication().getResourceBundle(
				context, "procMsg");
		return bundle.getString(key);
	}

	public void onUnselect(UnselectEvent event) {
		String userName = event.getObject().toString();
		GlobalAuthorsActivation.userList.remove(userName);
		if (GlobalAuthorsActivation.userList.size() != 1) {
			showPanel = false;
		} else {
			showPanel = true;
			onSelect(null);
		}
	}

	public void onSelect(SelectEvent event) {
		final String lBuf = "com.wissen.admin.model.UserActivation.init() :: ";
		logger.info(lBuf + WissenswerkConstants.INFO_ENTERED_METHOD);
		panelShow = true;
		String userName = null;
		Map<String, Map<String, List<String>>> myRoleRightsMap = new TreeMap<String, Map<String, List<String>>>();
		try {
			if (event != null) {
				userName = event.getObject().toString();
				if (!GlobalAuthorsActivation.userList.contains(userName)) {
					GlobalAuthorsActivation.userList.add(userName);
				}
			} else {
				userName = GlobalAuthorsActivation.userList.get(0);
			}

			if (GlobalAuthorsActivation.userList.size() != 1) {
				showPanel = false;
				return;
			} else {
				showPanel = true;

				IDfSession session = WissenswerkSessionManager.instance()
						.getWEBSession();

				AdminAPI adminApi = new AdminAPI();
				globalGroup = adminApi.getGlobalGroupName(userName, session);
				myRoleRightsMap = adminApi.getGroupName(userName, session);

				Iterator<Entry<String, Map<String, List<String>>>> itrDocType = myRoleRightsMap
						.entrySet().iterator();
				docTypeList = new ArrayList<DocTypeRoleBO>();
				while (itrDocType.hasNext()) {
					DocTypeRoleBO docTypeRoleBO = new DocTypeRoleBO();
					Entry<String, Map<String, List<String>>> myEntry = itrDocType
							.next();
					String area = myEntry.getKey();
					Map<String, List<String>> myRole = myEntry.getValue();
					Iterator<Entry<String, List<String>>> itrRoleType = myRole
							.entrySet().iterator();
					List<RoleBO> roleList = new ArrayList<RoleBO>();
					while (itrRoleType.hasNext()) {
						RoleBO roleBO = new RoleBO();
						Entry<String, List<String>> myRoleMap = itrRoleType
								.next();
						String role = myRoleMap.getKey();
						List<String> docListRoles = myRoleMap.getValue();
						List<String> docListRoles1 = new ArrayList<String>();
						if (role.equals("AREA_CO") || role.equals("RECON_CO")) {
							roleBO.setRoleName("aaaaaaaaaaaaaaa_CO_HEADER");
							docListRoles1.add(role);
							Collections.sort(docListRoles1);
							roleBO.setRoleList(docListRoles1);
						} else {
							roleBO.setRoleName(role);
							if (!docListRoles.isEmpty()) {
								Collections.sort(docListRoles);
								roleBO.setRoleList(docListRoles);
							}
						}
						roleList.add(roleBO);
					}
					docTypeRoleBO.setAreaName(area);
					Collections.sort(roleList, new Comparator<Object>() {
						public int compare(Object firstObject,
								Object secondObject) {
							if (firstObject != null && secondObject != null) {
								return firstObject
										.toString()
										.toLowerCase()
										.compareTo(
												secondObject.toString()
														.toLowerCase());
							} else {
								return 0;
							}
						}
					});
					docTypeRoleBO.setAreaList(roleList);
					docTypeList.add(docTypeRoleBO);
				}

				Collections.sort(docTypeList, new Comparator<Object>() {
					public int compare(Object firstObject, Object secondObject) {
						if (firstObject != null && secondObject != null) {
							return firstObject
									.toString()
									.toLowerCase()
									.compareTo(
											secondObject.toString()
													.toLowerCase());
						} else {
							return 0;
						}
					}
				});
			}
		} catch (AdminException exp) {
			logger.error(lBuf + exp);
		} catch (Exception exp) {
			logger.error(lBuf + exp);
		} finally {
			logger.info(lBuf + WissenswerkConstants.INFO_EXITED_METHOD);
		}
	}

	public void onclickPickList() {
		String value = FacesContext.getCurrentInstance().getExternalContext()
				.getRequestParameterMap().get("userNamePickListClick");
		GlobalAuthorsActivation.userList.clear();
		GlobalAuthorsActivation.userList.add(value);
		onSelect(null);
	}

}

Please let me know if you need any further information.

I am not able to setup this project on my system. I am getting below error while importing the project in Eclipse.

Could not calculate build plan: Plugin org.apache.maven.plugins:maven-war-plugin:3.1.0 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.maven.plugins:maven-war-plugin:jar:3.1.0
Plugin org.apache.maven.plugins:maven-war-plugin:3.1.0 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.maven.plugins:maven-war-plugin:jar:3.1.0

While executing the mvn command, I am getting below error.

pf-4909>mvn clean jetty:run-exploded
[INFO] Scanning for projects...
Downloading: https://artifactory.wob.vw.vwg:8443/artifactory/public/org/apache/maven/plugins/maven-war-plugin/3.1.0/maven-war-plugin-3.1.0.pom
[WARNING] Failed to retrieve plugin descriptor for org.apache.maven.plugins:maven-war-plugin:3.1.0: Plugin org.apache.maven.plugins:maven-war-plugin:3.1.0 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.maven.plugins:maven-war-plugin:jar:3.1.0
Downloading: https://artifactory.wob.vw.vwg:8443/artifactory/public/org/eclipse/jetty/jetty-maven-plugin/9.4.15.v20190215/jetty-maven-plugin-9.4.15.v20190215.pom
[WARNING] Failed to retrieve plugin descriptor for org.eclipse.jetty:jetty-maven-plugin:9.4.15.v20190215: Plugin org.eclipse.jetty:jetty-maven-plugin:9.4.15.v20190215 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.eclipse.jetty:jetty-maven-plugin:jar:9.4.15.v20190215
Downloading: https://artifactory.wob.vw.vwg:8443/artifactory/public/org/codehaus/mojo/maven-metadata.xml
Downloading: https://artifactory.wob.vw.vwg:8443/artifactory/public/org/apache/maven/plugins/maven-metadata.xml
[WARNING] Could not transfer metadata org.apache.maven.plugins/maven-metadata.xml from/to mweb-maven-mirror-public (https://artifactory.wob.vw.vwg:8443/artifactory/public): sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
[WARNING] Could not transfer metadata org.codehaus.mojo/maven-metadata.xml from/to mweb-maven-mirror-public (https://artifactory.wob.vw.vwg:8443/artifactory/public): sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.654 s
[INFO] Finished at: 2019-06-20T10:43:53+05:30
[INFO] Final Memory: 12M/153M
[INFO] ------------------------------------------------------------------------
[ERROR] No plugin found for prefix 'jetty' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (C:\Users\sdqy2od\.m2\repository), mweb-maven-mirror-public (https://artifactory.wob.vw.vwg:8443/artifactory/public)] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/NoPluginFoundForPrefixException

I am using my office laptop to setup the project provided by you and it might be have some restriction on dependency downloading. 

Are you able to replicate the issue?

--- a/src/main/resources/META-INF/resources/primefaces/picklist/picklist.js
+++ b/src/main/resources/META-INF/resources/primefaces/picklist/picklist.js
@@ -776,9 +776,9 @@ PrimeFaces.widget.PickList = PrimeFaces.widget.BaseWidget.extend({
                         $this.saveState();
                         $this.fireTransferEvent(items, from, to, type);
                     }
-
-                    $this.updateListRole();
                 });
+                
+                $this.updateListRole();
             });
         }
         else {

